[LibWebRTC] Include AVX2 sources on Intel hosts

dpino opened the pull request:
#### 0de2717c00294cf52a0f972df65a06cbadd65579
<pre>
[LibWebRTC] Include AVX2 sources on Intel hosts
<a href="https://bugs.webkit.org/show_bug.cgi?id=258708">https://bugs.webkit.org/show_bug.cgi?id=258708</a>

Reviewed by Carlos Alberto Lopez Perez.

This partially reverts 265646@main.

Include AVX2 sources if host is X86 or X64.
Build AVX2 sources with build flags &apos;-mavx2&apos; and &apos;-mfma&apos;.

* Source/ThirdParty/libwebrtc/CMakeLists.txt:
* Source/cmake/DetectAVX2.cmake: Removed.
* Source/cmake/WebKitCommon.cmake:

Canonical link: <a href="https://commits.webkit.org/265754@main">https://commits.webkit.org/265754@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4d4ee3b6f13e4d9d83b72cd0f7bbae2d3fb4e46f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/11792 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/11991 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/12361 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/13436 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/11231 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/14380 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/11972 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/14091 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/11956 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/12788 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/9999 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/13858 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/10080 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/10707 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/17835 "Passed tests") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/10025 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/11158 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/10862 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/14030 "Passed tests") | 
| [  ~~🛠 🧪 jsc-arm64~~](https://ews-build.webkit.org/#/builders/12/builds/11188 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/11268 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/9309 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/11875 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/10442 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/3193 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2840 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/14725 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/12210 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/11124 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/2959 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

clopez commented on Source/ThirdParty/libwebrtc/CMakeLists.txt:
```diff
@@ -1791,7 +1791,7 @@ if (WTF_CPU_X86_64 AND WTF_CPU_HAS_AVX2)
         Source/webrtc/modules/audio_processing/agc2/rnn_vad/vector_math_avx2.cc
     )
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=native -mno-avx512f")
+    set_source_files_properties(webrtc_SOURCES PROPERTIES COMPILE_OPTIONS "-mavx2;-mfma")
```
Instead of applying `set_source_files_properties` to all the `webrtc_SOURCES ` I think it would be better to create a new list named `webrtc_avx_SOURCES` or similar and then only apply this flags to that specific files.

Then the new list `webrtc_avx_SOURCES`  can be concatenated with the `webrtc_SOURCES ` one.

So the idea is to only build with compiler flags `"-mavx2;-mfma"` only this very specific source files related to `avx2`

clopez reviewed (approved):
LGTM! Thanks!

dpino committed:
[LibWebRTC] Include AVX2 sources on Intel hosts
https://bugs.webkit.org/show_bug.cgi?id=258708

Reviewed by Carlos Alberto Lopez Perez.

This partially reverts 265646@main.

Include AVX2 sources if host is X86 or X64.
Build AVX2 sources with build flags '-mavx2' and '-mfma'.

* Source/ThirdParty/libwebrtc/CMakeLists.txt:
* Source/cmake/DetectAVX2.cmake: Removed.
* Source/cmake/WebKitCommon.cmake:

Canonical link: https://commits.webkit.org/265754@main
